### PR TITLE
Fix exclusions of dependencies

### PR DESCRIPTION
**Pull request by anwar6953**
**TLDR: All exclusions for a particular dependency should be defined in one place, otherwise one set of exclusions may not be respected.**

[This PR](https://github.com/caskdata/cdap/pull/9891) added an exclusion to the grok dependency.
However, it then ignored the exclusion (on slf4j-log4j12) in the root pom.xml, and so `slf4j-log4j12` was being brought in.
Logging was then being messed up - integration tests logging were not being seen, for instance.

[This change](https://github.com/caskdata/cdap/commit/40d13e9eee4f5debdf0d676eb081a3c608bf5c4e#diff-121e2347eb44291a8c85846c63681454) does a similar thing - also fixed in this PR.

### CODE REVIEW

**Comment by anwar6953** [comment]
Without this change, I was seeing
```
SLF4J: Found binding in [jar:file:<HOME_DIR>/.m2/repository/org/slf4j/slf4j-log4j12/1.7.5/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
...
log4j:WARN No appenders could be found for logger (co.cask.cdap.test.IntegrationTestBase).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
```
in the integration test console output, and not seeing any other log messages.

With this change, I have verified that "Found binding" message for slf4j-log4j12-1.7.5.jar is gone, and I see logging in the console.

**Comment by chtyim** [comment]
LGTM

**Comment by anwar6953** [comment]
https://builds.cask.co/browse/CDAP-DUT6426-1

**Review by chtyim — APPROVED** [review]
